File: day-1/puzzle1.py

with open("puzzle-input.txt") as f:
    lines = f.readlines()

int_array = [int(s.strip()) for s in lines]

# part 1
diff_array = [a-b for a, b in zip(int_array[1:] + [int_array[-1]], int_array)]

inc_count = len(list(filter(lambda x: (x > 0), diff_array)))
print("Number of increases: ", inc_count)

# part 2

File: day-1/puzzle1.jl

f = open("day-1/puzzle-input.txt")
lines = readlines(f)

int_array = [parse(Int, x) for x in lines]

# part 1
diff_array = append!(int_array[2:end], int_array[end]) .- int_array

inc_count = sum(diff_array .> 0)
println("Number of increases: ", inc_count)

# part 2
window_array = int_array[1:end-2] .+ int_array[2:end-1] .+ int_array[3:end]
window_diff_array = append!(window_array[2:end], window_array[end]) .- window_array
inc_count = sum(window_diff_array .> 0)
println("Number of increases: ", inc_count)